src/cdef_apply.rs

```diff
@@ -3,268 +3,215 @@ use crate::include::common::intops::ulog2;
-                ),
+                &mut dst[0][y],
+                slice::from_raw_parts(&mut *src[0].offset(y_off + x_off as isize - 2), 2),
                 2,
             );
-            y += 1;
-            y_off += BD::pxstride(*src_stride.offset(0) as usize) as isize;
+            y_off += BD::pxstride(src_stride[0] as usize) as isize;
         }
     }
-    if layout as c_uint == Rav1dPixelLayout::I400 as c_int as c_uint
-        || flag as c_uint & BACKUP_2X8_UV as c_int as c_uint == 0
-    {
+    if layout == Rav1dPixelLayout::I400 || !flag.contains(Backup2x8Flags::BACKUP_2X8_UV) {
         return;
     }
-    let ss_ver = (layout as c_uint == Rav1dPixelLayout::I420 as c_int as c_uint) as c_int;
-    let ss_hor = (layout as c_uint != Rav1dPixelLayout::I444 as c_int as c_uint) as c_int;
+    let ss_ver = (layout == Rav1dPixelLayout::I420) as c_int;
+    let ss_hor = (layout != Rav1dPixelLayout::I444) as c_int;
     x_off >>= ss_hor;
     y_off = 0 as c_int as ptrdiff_t;
-    let mut y = 0;
-    while y < 8 >> ss_ver {
+    for y in 0..8 >> ss_ver {
         BD::pixel_copy(
-            &mut (*dst.offset(1))[y as usize],
-            slice::from_raw_parts(
-                &mut *(*src.offset(1)).offset((y_off + x_off as isize - 2 as isize) as isize)
-                    as *mut BD::Pixel,
-                2,
-            ),
+            &mut dst[1][y],
+            slice::from_raw_parts(src[1].offset(y_off + x_off as isize - 2), 2),
             2,
         );
         BD::pixel_copy(
-            &mut (*dst.offset(2))[y as usize],
-            slice::from_raw_parts(
-                &mut *(*src.offset(2)).offset((y_off + x_off as isize - 2 as isize) as isize)
-                    as *mut BD::Pixel,
-                2,
-            ),
+            &mut dst[2][y],
+            slice::from_raw_parts(src[2].offset(y_off + x_off as isize - 2), 2),
             2,
         );
-        y += 1;
-        y_off += BD::pxstride(*src_stride.offset(1) as usize) as isize;
+        y_off += BD::pxstride(src_stride[1] as usize) as isize;
     }
 }
 
 unsafe fn adjust_strength(strength: c_int, var: c_uint) -> c_int {
     if var == 0 {
-        return 0 as c_int;
+        return 0;
     }
+
     let i = if var >> 6 != 0 {
         cmp::min(ulog2(var >> 6), 12 as c_int)
     } else {
-        0 as c_int
+        0
     };
+
     return strength * (4 + i) + 8 >> 4;
 }
 
 pub(crate) unsafe fn rav1d_cdef_brow<BD: BitDepth>(
     c: &Rav1dContext,
-    tc: *mut Rav1dTaskContext,
-    p: *const *mut BD::Pixel,
+    tc: &mut Rav1dTaskContext,
+    p: &[*mut BD::Pixel; 3],
     lflvl: *const Av1Filter,
     by_start: c_int,
     by_end: c_int,
     sbrow_start: c_int,
     sby: c_int,
 ) {
-    let f: *mut Rav1dFrameContext = (*tc).f as *mut Rav1dFrameContext;
-    let bitdepth_min_8 = if 16 == 8 {
-        0 as c_int
+    let f: *mut Rav1dFrameContext = tc.f as *mut Rav1dFrameContext;
+    let bitdepth_min_8 = if 16 == 8 { 0 } else { (*f).cur.p.bpc - 8 };
```

Comment:
This was mis-translated by us when we did bitdepth deduplication.  It should be this:

```suggestion
    let bitdepth_min_8 = match BD::BPC {
        BPC::BPC8 => 0,
        BPC::BPC16 => (*f).cur.p.bpc - 8,
    };
```